Core-16349: corda 5 Kotlin negotiation app

7Dhruv opened the pull request.

7Dhruv committed:
Core-16349: corda 5 Kotlin negotiation app

7Dhruv committed:
Core-16349: corda 5 Kotlin negotiation app

tlawson3 commented on kotlin-samples/corda5-negotiation/contracts/build.gradle:
```diff
@@ -0,0 +1,84 @@
+    // The CorDapp uses the slf4j logging framework. Corda-API provides this so we need a 'cordaProvided' declaration.
+    cordaProvided 'org.slf4j:slf4j-api'
+
+    // This are shared so should be here.
```
This block should probably be deleted

tlawson3 commented on kotlin-samples/corda5-negotiation/workflows/build.gradle:
```diff
@@ -0,0 +1,85 @@
+    // The CorDapp uses the slf4j logging framework. Corda-API provides this so we need a 'cordaProvided' declaration.
+    cordaProvided 'org.slf4j:slf4j-api'
+
+    // Dependencies Required By Simulator Tests
```
to remove

peterli-r3 reviewed (changes requested):
Umm, where are the gradle files, and config file for deployment?

7Dhruv committed:
Merge remote-tracking branch 'origin/Dhruv/Core-16349_negotiation_kt' into Dhruv/Core-16349_negotiation_kt

7Dhruv committed:
- Refactored gradle files

7Dhruv committed:
Merge remote-tracking branch 'origin/Dhruv/Core-16349_negotiation_kt' into Dhruv/Core-16349_negotiation_kt

7Dhruv commented:
> Umm, where are the gradle files, and config file for deployment?

- Included them in the new commit

peterli-r3 commented:
![image](https://github.com/corda/corda5-samples/assets/51169685/7a0a4a91-9462-4e72-98c6-fa48d232acdb)
Something is not right. You have long hanging flow. All the function works, but it seems there is continuous running flows.

7Dhruv commented:
> ![image](https://user-images.githubusercontent.com/51169685/263931068-7a0a4a91-9462-4e72-98c6-fa48d232acdb.png) Something is not right. You have long hanging flow. All the function works, but it seems there is continuous running flows.

Do you mean that the flow is taking too long to execute? and does this happen in java version of this app as well or is it just this one?

khutsijabari commented on kotlin-samples/corda5-negotiation/workflows/src/main/kotlin/com/r3/developers/samples/negotiation/workflows/Propose/ProposalFlowRequest.kt:
```diff
@@ -0,0 +1,93 @@
+    var flowEngine: FlowEngine? = null
+
+    @Suspendable
+    override fun call(requestBody: ClientRequestBody): String {
```
Why does ProposalFlow return a string? Is it not supposed to return a UUID? Taking into account that Accept and Modify flows seem to be expecting the proposal ID to be a UUID

7Dhruv committed:
- Refactored variable names
- updated accept contract verification
- Resolved hanging flows bug
- updated readme.md

peterli-r3 commented on kotlin-samples/corda5-negotiation/README.md:
```diff
@@ -0,0 +1,125 @@
+  "clientRequestId": "AcceptFlow",
+  "flowClassName": "com.r3.developers.samples.negotiation.workflows.Accept.AcceptFlowRequest",
+  "requestBody": {
+      "proposalID": "<use the proposal id here>"
```
You are missing the variable acceptor, but why do you need this input? 
The accepter can only be the proposee of the proposal. In the contract, you can just use the proposee variable.

khutsijabari replied:
Hi @peterli-r3 I've added the Acceptor to the documentation. We still need to know who is trying to accept the trade to be able to perform the necessary checks in the Accept contract.

peterli-r3 replied:
Hi @khutsijabari , I had quickly chatted with @7Dhruv eariler, You dont really need to add a variable to keep track of the proposer. You can check it in two ways: 
1. Check if the proposer with myInfo in the AcceptProposal flow initiator. 
2. Make the accept command take input, we can pass in the myInfo into the contract via the command. 
The idea is that we will try to have as less variables as possible in the states.
